### PR TITLE
Add in the ability to get thumbnails for VideoPress via media endpoint.

### PR DESCRIPTION
**Pull request by dbtlr**
When we pushed out VideoPress, it was missing poster thumbnails. This is needed to make Calypso work correctly with Jetpack sites.

This PR seeks to add thumbnails by adding the wpcom functions that the API tries to call, to add in the images.

Request: Can we get this in the next point release for Jetpack, to make sure we get Calypso working ASAP?

## Testing

Simply check the media library in Calypso and make sure thumbnails are loading.

If checking the output for the media API for a Jetpack site, you should see the thumbnails reporting in there like this:

screen shot 2017-03-10 at 11 49 22 pm

### CODE REVIEW

File: modules/videopress/class.videopress-xmlrpc.php [hunk 1]
```diff
@@ -155,18 +155,25 @@ public function update_poster_image( $request ) {
 			return false;
 		}
 
+		// Switch to photon if needed.
+		$poster = str_replace(
+			'https://videos.files.wordpress.com',
+			'https://i0.wp.com/videos.files.wordpress.com',
```
Comment (jeherve) [inline modules/videopress/class.videopress-xmlrpc.php]: Should we use `apply_filters( 'jetpack_photon_url`, $poster )` here, so all Photon filters / domain sharding / etc can be applied if needed?

File: modules/videopress/utility-functions.php [hunk 2]
```diff
@@ -558,21 +558,145 @@ function videopress_make_media_upload_path( $blog_id ) {
...
+	if ( is_wp_error( $post ) ) {
+		return null;
+	}
+
+	$meta = wp_get_attachment_metadata( $post->ID );
+
+	// Switch to photon if needed.
+	$poster = str_replace(
+		'https://videos.files.wordpress.com',
+		'https://i0.wp.com/videos.files.wordpress.com',
```
Comment (jeherve) [inline modules/videopress/utility-functions.php]: Should we use the `jetpack_photon_url` filter here as well?

File: modules/videopress/class.videopress-xmlrpc.php (hunk 1 above)
Comment (dbtlr) [inline modules/videopress/class.videopress-xmlrpc.php]: Makes sense, I'll apply this change and test. I was really just trying to fix the fact that Calypso won't work without Photon URLs here. But it makes sense to use the internal methods for this.

File: modules/videopress/utility-functions.php (hunk 2 above)
Comment (dbtlr) [inline modules/videopress/utility-functions.php]: Same

**Comment by dbtlr** [comment]
@jeherve I've made the requested changes and with a couple tiny gotchas, it seems to be working fine now.

**Comment by dereksmart** [comment]
merged to 4.7 ebcd7c1

**Pull request merged**